New parameter: weeksToShow

tricasthu opened the pull request:
I have added an optional parameter: weeksToShow.

First, let me explain the use case for this.
I want to use this calendar in the way it already is, a full-screen, Google Calendar-like widget, and it is excellent for that.
However, I also want to show a "reduced" version of the full calendar, where I can show an event and a few weeks before and after. Think of that like looking at the calendar through a window.

To be able to do that, I have decided to use the number of weeks to show instead of eg. providing an initial display date since then I would have had to deal with situations eg. the given date is not a Monday or Sunday or whatever the first day of the week is.

I tried my best to sanitize the user input while providing parameter documentation as well.
I also tried to keep code changes to a minimum.
I have corrected a typo in one of the constant members as well (kMaxWeekPerMoth -> kMaxWeekPerMonth).

balazsbokanyi committed:
added weeksToShow parameter

balazsbokanyi committed:
added localized week days builder

tricasthu commented:
@NikitaAmelinCR @NikitaAmelinUnikoom @BogdanPushkarCR @jekaUA 

Just a friendly ping trying to get your attention on these PRs. Is this package still supported? I would like to contribute but do not want to wait weeks for a feature to be released as well. If this package is discontinued, I can create my own copy of the repo as per your license if I'm not mistaken.

tricasthu committed:
Merge pull request #1 from tricastdev/feature/add-localized-day-of-week

added localized week days builder

tricasthu committed:
Merge branch 'main' into feature/number-of-weeks-in-view

NikitaAmelinCR commented:
Hi! Sorry for such delay. We still support this package.
I left a comment and also noticed an issue.
If a user adds `localizedWeekDaysBuilder` to CrCalendar and uses setState to rebuild it, then the first day of the week is determined by the app's current locale, but calendar grid is not changed accordingly. Can you fix it please?

balazsbokanyi committed:
modified example to not use a const list

tricasthu commented on example/lib/pages/calendar_page.dart:
```diff
@@ -109,6 +109,8 @@ class _CalendarPageState extends State<CalendarPage> {
               onDayClicked: _showDayEventsInModalSheet,
               minDate: DateTime.now().subtract(const Duration(days: 1000)),
               maxDate: DateTime.now().add(const Duration(days: 180)),
+              //weeksToShow: const [0,1,2],
```
I modified the example, I hope that's what you meant. If not, please can you give me more info on what would you like me to do?

NikitaAmelinCR replied:
yes, that's what I meant. Thanks!

tricasthu commented:
I could not reproduce the issue with `localizedWeekDaysBuilder`. Can you give me an example?

I have an example app, and I have added internationalization to it. I have also added that pressing a button changes the current language (simulating a real app language change) with setState and both the weekday abbreviations and the grid changed accordingly.

NikitaAmelinCR commented:
sure.
<img width="488" alt="image" src="https://github.com/Cleveroad/cr_calendar/assets/83710983/5abe1984-a13c-4939-accc-d691a15aa63b">
and use setState to change the condition
<img width="298" alt="image" src="https://github.com/Cleveroad/cr_calendar/assets/83710983/815d1a89-6919-473f-ab3f-9da8a5afdb5a">

tricasthu commented:
OK, got it. I initialized the firstDayOfWeek in didChangeDependencies, however, that does not run on every build, therefore the first day of the week did not change. I [added the fix](https://github.com/Cleveroad/cr_calendar/pull/22/commits/b33b284f706fa9b4d6715a578c377fabdff1e223).

tricasthu commented:
@NikitaAmelinCR are there any more issues or changes I should make?

NikitaAmelinCR commented:
@tricasthu 
Hi! No, all is good, thanks. I merged features.
